tests/testMongo.js

```diff
@@ -0,0 +1,154 @@
+'use strict';
+const BaseController = require('../api/base-controller');
+const MongoDbModel = require('../api/mongodb-model');
+const HappyCrud = require('../api');
+const BaseTest = require('./base-test');
+const should = require('should');
+const mongoose = require('mongoose');
+
+class ControllerB extends BaseController {
+} // class ModelA
+
+class TestB extends BaseTest {
+
+constructor(server) {
+  super(server);
+
+  const schema = {
+    a: String,
+    b: String,
+    c: Number,
+  }
+
+  const options = {
+    word: 'b',
+    path: '/api',
+    mount: '/',
+  }
+
+  const col = 'testb';
+  const model = new MongoDbModel(col, schema);
+  const ctrl = new ControllerB(model);
+  const api = new HappyCrud(server, ctrl, options);
+
+  // Setup db and return instance in promise
+  return new Promise((resolve) => {
```

Comment:
Let's try something else with event emission. But as this could happen instantly, try to use the event system which supports pending events.